src/core/simulation_stages.jl

```diff
@@ -149,13 +140,13 @@ function build!(
     horizon::Int,
     stage_interval::Dates.Period,
 )
+    stage_built(stage) && reset!(stage)
     settings = get_settings(get_psi_container(stage))
     set_horizon!(settings, horizon)
```

Comment:
It's not obvious why you are setting horizon and initial_time in `settings`.  A comment would help.